apisix/stream/router/ip_port.lua

```diff
@@ -71,6 +76,18 @@ do
             end
 
             local route = item.value
+            if item.value.remote_addr then
+                local remote_matcher = core_ip.create_ip_matcher({item.value.remote_addr})
+                if remote_matcher then
```

Comment:
We don't need to check the returned value, it should be checked in the check function.

Reply:
The check process in the check function just checks the address if server_addr or remote_addr is validated.

But in some circumstances, both the remote_address and server_address are not existed, but the configuration is validated. Here's the example in https://github.com/apache/apisix/blob/master/docs/zh/latest/stream-proxy.md#%E6%8E%A5%E6%94%B6-tls-over-tcp

```bash
curl http://127.0.0.1:9080/apisix/admin/stream_routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
{
    "sni": "a.test.com",
    "upstream": {
        "nodes": {
            "127.0.0.1:5991": 1
        },
        "type": "roundrobin"
    }
}'
```
So we can use the nil-check code here to avoid a useless matcher-created processes.

Reply:
I mean the returned value of create_ip_matcher